### PR TITLE
Forward track-matching workflow

### PR DESCRIPTION
**Pull request by rpezzi**
This PR adds the `o2-globalfwd-matcher-workflow` to match MFT and MCH tracks

Added:
* GlobalFwdTrack dataformat: Base format for GlobalMuonTracks (MFT+MCH+MID) and others
* GlobalForward matcher, track-writer, AOD producer workflow

Eventually the MCH-MID match will be used by `o2-globalfwd-matcher-workflow` to produce both MFT-MCH (a.k.a GlobalFwdTracks) and MFT-MCH-MID (a.k.a. GlobalMuonTracks).

### CODE REVIEW

**Comment by rpezzi** [comment]
FullCI failed due to missing `globalfwd-track-matcher` on `full-system-test/dpl-workflow.sh`. This has been added in the lastest commit, but pipeline size N_MFTMCH is not adapted. @davidrohr , can you please check [here](https://github.com/AliceO2Group/AliceO2/pull/6966/files#diff-95b1619e66e8d9d1a3ff6fe786dfc8d86a63f79be0d9859d28618579e70da071R116)? Thanks,
Rafael

File: prodtests/full-system-test/dpl-workflow.sh [hunk 1]
```diff
@@ -196,6 +197,7 @@ fi
 has_detector ITS && WORKFLOW+="o2-its-reco-workflow $ARGS_ALL --trackerCA $DISABLE_MC --clusters-from-upstream --disable-root-output $ITS_CONFIG --configKeyValues \"$ARGS_ALL_CONFIG;$ITS_CONFIG_KEY\" --its-dictionary-path $FILEWORKDIR --pipeline its-tracker:$N_ITSTRK | "
 has_detector TPC && WORKFLOW+="o2-gpu-reco-workflow ${ARGS_ALL//-severity $SEVERITY/-severity $SEVERITY_TPC} --input-type=$GPU_INPUT $DISABLE_MC --output-type $GPU_OUTPUT --pipeline gpu-reconstruction:$N_TPCTRK $GPU_CONFIG --configKeyValues \"$ARGS_ALL_CONFIG;GPU_global.deviceType=$GPUTYPE;GPU_proc.debugLevel=0;$GPU_CONFIG_KEY;$GPU_EXTRA_CONFIG\" | "
 has_detectors ITS TPC && WORKFLOW+="o2-tpcits-match-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output $DISABLE_MC --its-dictionary-path $FILEWORKDIR --pipeline itstpc-track-matcher:$N_TPCITS | "
+has_detectors MFT MCH && WORKFLOW+="o2-globalfwd-matcher-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output $DISABLE_MC --pipeline globalfwd-track-matcher:$N_MFTMCH | "
```
Comment (shahor02) [inline prodtests/full-system-test/dpl-workflow.sh]: @rpezzi you are adding MFT/MCH matching to common part of the workflow (i.e. running both in sync. and async versions) while the MFT and MCH reconstructions are enabled only in the async mode. Could you move this line to the block `if [ $SYNCMODE == 0 ]; then`, e.g. after https://github.com/AliceO2Group/AliceO2/blob/092dc6f24fe79fcec3612ce1f3679790537b228c/prodtests/full-system-test/dpl-workflow.sh#L211 ?

Comment (rpezzi) [reply prodtests/full-system-test/dpl-workflow.sh]: Thanks. Done.

**Comment by rpezzi** [comment]
Hello experts, 
I count with your support to proceed with this PR, which was rebased to fix merge conflicts a couple of times. @aphecetche @bovulpes  @shahor02 , could you kindly take a look at this code? I appreciate if non-critical changes could be addressed in forthcoming PRs.
Thanks,

File: DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainer.h [hunk 2]
```diff
@@ -409,6 +415,11 @@ struct RecoContainer {
   auto getTPCITSTracksMCLabels() const { return getSpan<o2::MCCompLabel>(GTrackID::ITSTPC, MCLABELS); }
   auto getTPCITSTrackMCLabel(GTrackID id) const { return getObject<o2::MCCompLabel>(id, MCLABELS); }
 
+  // MFT-MCH
+  const o2::dataformats::GlobalFwdTrack& getGlobalFwdTrack(GTrackID gid) const { return getTrack<o2::dataformats::GlobalFwdTrack>(gid); }
```
Comment (aphecetche) [inline DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainer.h:419]: I wonder if/how the naming will change when the MID matching will be introduce (i.e. when we'll get MFT-MCH-MID tracks). Any thoughts on that ?

Comment (rpezzi) [reply DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainer.h:419]: FwdTrack is already taken by the base class, so the choice of global here was to be generic. The identification of GlobalMuonTrack to be performed by the user of `getGlobalFwdTrack` by reading a positive MIDmatchChi2, most notably to select the track-type at AOD production.

**Review by shahor02 — COMMENTED** [review]
Hi @rpezzi 

Looked only superficially, looks fine, right now cannot really check all changes. 
Why do you use double for such things like chi2?
For the naming, why not using explicit names getTrackMFTMCH and getTrackMFTMCHMID?

Could you add this workflow to sim_challenge.sh, and run it locally? If everything works including AOD creation, we can merge it.

**Comment by rpezzi** [comment]
> Why do you use double for such things like chi2?

There is room for optimization

> For the naming, why not using explicit names getTrackMFTMCH and getTrackMFTMCHMID?

The next step is to include MID here, then the output will include both MFTMCH and MFTMCHMID.

> Could you add this workflow to sim_challenge.sh, and run it locally? If everything works including AOD creation, we can merge it.

This workflow is already on It is already on sim_challenge.sh [(here)](https://github.com/AliceO2Group/AliceO2/pull/6966/files#diff-29b216e4f5f241b2b4ee46efc24ad7dc8f13b0c85105a5eab8665a315df8e4d5R181). 
AOD production tested locally. Tracktype 2 = GlobalMuonTrack; TrackType 4 = MCHStandalone.
image

**Comment by shahor02** [comment]
> This workflow is already on It is already on sim_challenge.sh [(here)](https://github.com/AliceO2Group/AliceO2/pull/6966/files#diff-29b216e4f5f241b2b4ee46efc24ad7dc8f13b0c85105a5eab8665a315df8e4d5R181).
> AOD production tested locally. Tracktype 2 = GlobalMuonTrack; TrackType 4 = MCHStandalone.

sorry, did not notice. Fine for me, if it is ok for MFT and Moun, I will merge it.

**Review by shahor02 — APPROVED** [review]

**Review by bovulpes — APPROVED** [review]

**Review by aphecetche — APPROVED** [review]

**Pull request merged**